tests/integration_tests/conftest.py

```diff
@@ -260,3 +285,72 @@ async def connect_and_sign_in(
     hello = await read_until_command(proto, "welcome", timeout=120)
     player_id = hello["id"]
     return player_id, session, proto
+
+
+@pytest.fixture
+async def channel():
+    connection = await aio_pika.connect(
+        "amqp://{user}:{password}@localhost/{vhost}".format(
+            user=config.MQ_USER,
+            password=config.MQ_PASSWORD,
+            vhost=config.MQ_VHOST
+        )
+    )
+    channel = await connection.channel()
+
+    yield channel
+
+    await connection.close()
+
+
+async def connect_mq_consumer(server, channel, routing_key):
+    """
+    Returns a subclass of Protocol that yields messages read from a rabbitmq
+    exchange.
+    """
+    exchange = await channel.declare_exchange(
+        config.MQ_EXCHANGE_NAME,
+        aio_pika.ExchangeType.TOPIC
+    )
+    queue = await channel.declare_queue("", exclusive=True)
+    await queue.bind(exchange, routing_key=routing_key)
+    proto = AioQueueProtocol(queue)
+    await proto.consume()
+
+    if hasattr(server, "__connected_client_protos"):
+        server.__connected_client_protos.append(proto)
+
+    return proto
+
+
+class AioQueueProtocol(Protocol):
+    """
+    A wrapper around an asyncio `Queue` that exposes the `Protocol` interface.
+    """
+
+    def __init__(self, queue):
+        super().__init__(mock.Mock(), mock.Mock())
```

Comment:
Is it work autospeccing anything for either of these mocks?

Reply:
I don't think so. Now that I think of it though. I don't think I need to call `super` here at all, I think the super call is totally optional in python, and I don't need any of the base class behavior here.

Reply:
It is optional, yep.